source/grid/manifold_lib.cc

```diff
@@ -178,6 +152,78 @@ SphericalManifold<dim,spacedim>::push_forward_gradient(const Point<spacedim> &sp
 }
 
 // ============================================================
+// SphericalManifold
+// ============================================================
+
+template <int dim, int spacedim>
+SphericalManifold<dim,spacedim>::SphericalManifold(const Point<spacedim> center):
+  center(center)
+{}
+
+template <int dim, int spacedim>
+Point<spacedim>
+SphericalManifold<dim,spacedim>::
+get_new_point (const Point<spacedim> &P1, const Point<spacedim> &P2, double w) const
+{
+  if ( P1 == P2 )
+    return P1;
+
+  if (dim == spacedim)
+    {
+      return Point<spacedim>((1-w) * P1 + w * P2);
+    }
+  else
+    {
+      Assert(w >=0.0 && w <= 1.0,
+             ExcMessage("w should be in the range [0.0,1.0]."));
+      Assert((center-P1).norm() == (center-P2).norm(),
+             ExcMessage("P1 and P2 should have the same norm."));
+
+      Tensor<1,spacedim> V1 = P1 - center;
+      Tensor<1,spacedim> V2 = P2 - center;
+      double R = V1.norm();
+      double gamma = 2 * std::asin( P2.distance(P1) / (2*R) );
+      double sigma = w * gamma;
+      Tensor<1,spacedim> n = V2/R - ((V2/R)*(V1/R))*(V1/R);
+      n = n/n.norm();
+      Tensor<1,spacedim> V = std::cos(sigma) * V1 + std::sin(sigma) * R * n;
```

Comment:
I added some documentation.
This should be valid for all dimension. Basically I am looking for the maximum circumference  passing through P1 and P2. (This choice is obvious in 2D. )

> On 12 Jun 2016, at 16:04, Wolfgang Bangerth <notifications@github.com> wrote:
> 
> In source/grid/manifold_lib.cc <https://github.com/dealii/dealii/pull/2676#discussion_r66721302>:
> 
> > +    }
> > +  else
> > +    {
> > +      Assert(w >=0.0 && w <= 1.0,
> > +             ExcMessage("w should be in the range [0.0,1.0]."));
> > +      Assert((center-P1).norm() == (center-P2).norm(),
> > +             ExcMessage("P1 and P2 should have the same norm."));
> > +
> > +      Tensor<1,spacedim> V1 = P1 - center;
> > +      Tensor<1,spacedim> V2 = P2 - center;
> > +      double R = V1.norm();
> > +      double gamma = 2 * std::asin( P2.distance(P1) / (2*R) );
> > +      double sigma = w * gamma;
> > +      Tensor<1,spacedim> n = V2/R - ((V2/R)*(V1/R))*(V1/R);
> > +      n = n/n.norm();
> > +      Tensor<1,spacedim> V = std::cos(sigma) * V1 + std::sin(sigma) * R * n;
> with the exception of n, you can make all of these local variables const.
> 
> Can you add a comment that explains the formulas here? Is it valid for all dimensions/space dimensions?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/dealii/dealii/pull/2676/files/942343e92542be243ebab1f8c395cf4d52daa494#r66721302>, or mute the thread <https://github.com/notifications/unsubscribe/AIisRVdwIH_KKOcSH5hRGHbc-myG2Draks5qLBHlgaJpZM4IyZF9>.
>